src/main/java/frc/robot/robot_manager/RobotManager.java

```diff
@@ -49,102 +49,90 @@ protected void collectInputs() {}
 
   @Override
   protected RobotState getNextState(RobotState currentState) {
-    switch (currentState) {
+    return switch (currentState) {
       case SPEAKER_WAITING,
-          AMP_WAITING,
-          FEEDING_WAITING,
-          PASS_WAITING,
-          IDLE_NO_GP,
-          IDLE_WITH_GP,
-          CLIMBING_1_LINEUP,
-          CLIMBING_2_HANGING -> {}
-      case SPEAKER_SCORING -> {
-        if (!queuer.hasNote()) {
-          setStateFromRequest(RobotState.IDLE_NO_GP);
-        }
-      }
-      case AMP_SCORING -> {
-        if (!queuer.hasNote()) {
-          setStateFromRequest(RobotState.IDLE_NO_GP);
-        }
-      }
-      case FEEDING_SHOOTING -> {
-        if (!queuer.hasNote()) {
-          setStateFromRequest(RobotState.IDLE_NO_GP);
-        }
-      }
-      case PASS_SHOOTING -> {
-        if (!queuer.hasNote()) {
-          setStateFromRequest(RobotState.IDLE_NO_GP);
-        }
-      }
-      case SPEAKER_PREPARE_TO_SCORE -> {
-        if (shooter.atGoal() && arm.atGoal()) {
-          setStateFromRequest(RobotState.SPEAKER_SCORING);
-        }
-      }
+              AMP_WAITING,
+              FEEDING_WAITING,
+              PASS_WAITING,
+              IDLE_NO_GP,
+              IDLE_WITH_GP,
+              CLIMBING_1_LINEUP,
+              CLIMBING_2_HANGING ->
+          currentState;
+      case SPEAKER_SCORING, AMP_SCORING, FEEDING_SHOOTING, PASS_SHOOTING ->
+          !queuer.hasNote() ? RobotState.IDLE_NO_GP : currentState;
```

Comment:
a bit of a nitpick but this can be written without the `!` like this:
```suggestion
          queuer.hasNote() ? currentState : RobotState.IDLE_NO_GP;
```